### PR TITLE
feat: Improve table of contents

### LINKED ISSUE #173: Improve the table of content
Our table of content (left side) is quite poor and would deserve enhancements.
For instance:
1. Better highlight of the current selected component (may impact also the header)
2. Better identification of a main section with icons (see [MuleSoft](https://docs.mulesoft.com/general/))
3. Better highlight of the current selected page (see [Spring](https://docs.spring.io/spring-security/reference/migration/servlet/session-management.html), [RedHat Vale](https://redhat-documentation.github.io/vale-at-red-hat/docs/main/user-guide/using-vale-app/))
4. Separation between main sections (see [Hazelcast](https://docs.hazelcast.com/cloud/developer-guide), [Rudder](https://docs.rudder.io/get-started/current/advanced-configuration/index.html))

### PR DESCRIPTION
**Pull request by jeromecambon**
closes #173

### CODE REVIEW

**Review by tbouffard — APPROVED** [review]
LGTM

image

image

File: src/stylesheets/nav.scss [hunk 1]
```diff
@@ -255,18 +254,43 @@ html.is-clipped--nav {
...
+//.is-current-page {
+//  > .nav-link {
+//    color: var(--color-higlight-link);
+//  }
+//
+//  > .nav-text {
+//    font-weight: var(--body-font-weight-bold);
+//  }
+//}
+
```
Comment (tbouffard) [inline src/stylesheets/nav.scss:286]:
```suggestion
```

**Comment by tbouffard** [comment]
With the last title size increase

before | now
---- | -----
image | image

File: src/partials/nav-menu.hbs [hunk 2]
```diff
@@ -2,7 +2,7 @@
 <div class="nav-panel-menu is-active" data-panel="menu">
   <nav class="nav-menu">
     {{#with @root.page.componentVersion}}
-    <h3 class="title"><a href="{{{relativize ./url}}}">{{./title}}</a></h3>
+    <h3 class="title">{{./title}}</h3>
```
Comment (tbouffard) [inline src/partials/nav-menu.hbs:5]: This removes a link in the navbar. I don't understand why it should be removed


image
_previous navigation facilities_

Comment (jeromecambon) [reply src/partials/nav-menu.hbs:5]: I made the (bad) assumption it was not useful.

File: src/partials/nav-menu.hbs [hunk 3]
```diff
@@ -2,7 +2,7 @@
 <div class="nav-panel-menu is-active" data-panel="menu">
   <nav class="nav-menu">
     {{#with @root.page.componentVersion}}
-    <h3 class="title"><a href="{{{relativize ./url}}}">{{./title}}</a></h3>
+      <h3 class="title"><a href="{{{relativize ./url}}}">{{./title}}</a></h3>
```
Comment (tbouffard) [inline src/partials/nav-menu.hbs:5]: let's revert extra change not related to the topic
```suggestion
    <h3 class="title"><a href="{{{relativize ./url}}}">{{./title}}</a></h3>
```

**Pull request merged**